include/arch/riscv/arch/object/objecttype.h

```diff
@@ -43,5 +44,25 @@ static inline CONST bool_t Arch_isIRQControlDescendant(cap_t cap)
  */
 static inline CONST bool_t Arch_isMDBParentOf(cap_t cap_a, cap_t cap_b, bool_t firstBadged)
 {
-    return true;
+    switch (cap_get_capType(cap_a)) {
+#ifdef CONFIG_ALLOW_SBI_CALLS
+    case cap_sbi_cap: {
+        word_t eid_a_badge = cap_sbi_cap_get_capSBIEIDBadge(cap_a);
+        word_t eid_b_badge = cap_sbi_cap_get_capSBIEIDBadge(cap_b);
+        word_t fid_a_badge = cap_sbi_cap_get_capSBIFIDBadge(cap_a);
+        word_t fid_b_badge = cap_sbi_cap_get_capSBIFIDBadge(cap_b);
+
+        if (eid_a_badge != eid_b_badge) {
+            return false;
+        }
+
+        return (eid_a_badge == eid_b_badge && fid_a_badge == fid_b_badge && !firstBadged);
```

Comment:
This is missing the unbadged cap_a case. I think the above if-check should be `if (!eid_a) return true;`.

Also, checking just `!firstBadged` would probably be sufficient. But if you do want to compare all fields, you could do a structure compare: `return cap_a == cap_b && !firstBadged;`. I guess the CParser doesn't support that though.